### PR TITLE
Build: migrate to com.gradleup.shadow

### PR DESCRIPTION
**Pull request by Goooler**
https://github.com/GradleUp/shadow/releases/tag/8.3.0

Follow up:

- 068779f850fe2aff41fd2b18f5f425a075c34d0c
- c2f498cbcecfabb886c81bfa11a8ce6d6b5cbcb1

### CODE REVIEW

**Comment by Goooler** [comment]
CC @Badya

File: kotlin-native/gradle.properties [hunk 1]
```diff
@@ -30,7 +30,6 @@ org.gradle.jvmargs='-Dfile.encoding=UTF-8'
 org.gradle.workers.max=4
 slackApiVersion=1.2.0
 ktorVersion=1.2.1
-shadowVersion=8.1.1
```
Comment (Goooler) [inline kotlin-native/gradle.properties:33]: Seems this is missed in 068779f850fe2aff41fd2b18f5f425a075c34d0c.

**Comment by Badya** [comment]
@Goooler, thanks for the contribution and your hard work maintaining the shadow plugin!

**Comment by Goooler** [comment]
@Badya CC

We land it!

https://github.com/GradleUp/shadow/releases/tag/8.3.0

**Comment by Badya** [comment]
I'll run all the necessary CI checks and merge if everything passes

File: gradle/verification-metadata.xml [hunk 2]
```diff
@@ -1112,6 +1112,14 @@
             <sha256 value="298f0c2e979532171a05a41603263733b65f89113d4e809ccd0e26990149de18" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="com.gradleup.shadow" name="shadow-gradle-plugin" version="8.3.0">
+         <artifact name="shadow-gradle-plugin-8.3.0.jar">
+            <md5 value="d481a72bf8ce2d0e0a3b15d9190518c1" origin="Generated by Gradle"/>
+            <sha256 value="ffcb98f1d14698ee45e3aad02755281bea0617d3f8d3ae20047f54ad9c4f5a4b" origin="Generated by Gradle">
```
Comment (Badya) [inline gradle/verification-metadata.xml:1118]: This hash points to another artifact causing CI failures and should be updated accordingly to [doc](https://github.com/JetBrains/kotlin/tree/master?tab=readme-ov-file#dependency-verification)

**Review by Badya — CHANGES_REQUESTED** [review]

File: gradle/verification-metadata.xml (hunk 2 above)
Comment (Goooler) [reply gradle/verification-metadata.xml]: Thanks for the heads-up! Done.

**Comment by Badya** [comment]
CI build passed

Thanks for contribution!

**Comment by Goooler** [comment]
Thanks for the review!

**Pull request merged**